### PR TITLE
Admin users

### PR DESCRIPTION
**Pull request by maykremer**

### CODE REVIEW

File: GenerateUser/index.ts [hunk 1]
```diff
@@ -46,8 +46,13 @@ const generateUser: AzureFunction = async (context: Context, req: HttpRequest, r
...
+        const isRoleAdmin = [Role.Mada, Role.Technical, Role.Itur].includes(role);
 
-        if (!doesUnitExist(unitId)) {
+        if (isRoleAdmin && unitId) {
+            throw new ValidationFunctionError(functionName, `User with role ${role} can't have a unit`);
+        }
+
+        if (!isRoleAdmin && !doesUnitExist(unitId)) {
             throw new ValidationFunctionError(functionName, `Unit ${unitId} does not exist`);
         }
```
Comment (carmeltsadok) [inline GenerateUser/index.ts:57]:
```suggestion
 if (isRoleAdmin) {
            if (unitId) {
                throw new ValidationFunctionError(functionName, `User with role ${role} can't have a unit`);
            }
        } else if (!doesUnitExist(unitId)) {
            throw new ValidationFunctionError(functionName, `Unit ${unitId} does not exist`);
        }
```

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — APPROVED** [review]

File: GetUsersByRoles/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { Context, HttpRequest } from '@azure/functions';
+import { ValidationError } from 'joi';
+import authWrapper from '../shared/authWrapper';
+import { ForbiddenFunctionError, httpErrorHandler, ValidationFunctionError } from '../shared/error/error';
+import getConnection from '../shared/mongoose';
+import { Role, User } from '../shared/user/user.interface';
+import userModel from '../shared/user/user.model';
+import getUsersByRolesSchema from './joi';
+
+const functionName = 'getUsersByRoles';
```
Comment (lustigy96) [inline GetUsersByRoles/index.ts:10]: Add this function to the API

File: GetUsersByRoles/joi.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import * as Joi from 'joi';
+
+export const getUsersByRolesSchema: Joi.ObjectSchema = Joi.object({
+    body: Joi.object()
+        .keys({
+            roles: Joi.array().items(Joi.string()),
```
Comment (lustigy96) [inline GetUsersByRoles/joi.ts:6]: you can validate that the role is from the enum Role (I did that in the node-type and in the questions-type, so there is an example there)

File: GenerateUser/joi.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ export const generateUserSchema: Joi.ObjectSchema = Joi.object({
     body: Joi.object()
         .keys({
             name: Joi.string().min(2).required(),
-            unitId: Joi.string().min(2).required(),
+            unitId: Joi.string().min(2),
```
Comment (lustigy96) [inline GenerateUser/joi.ts:8]: you can check 24 chars since it is an object ID

**Review by lustigy96 — CHANGES_REQUESTED** [review]

**Pull request merged**